### PR TITLE
Use compare methods instead of test_utilities' .

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to #242 

## Summary
 - Do not use `maliput::test_utilities`'s compare methods.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: src/maliput_malidrive/test_utilities/assert_compare.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// BSD 3-Clause License
```
Comment (agalbachicar) [inline src/maliput_malidrive/test_utilities/assert_compare.h:1]: What if we move it to the test folder so there is no doubt about this file being used in production libraries?

**Review by agalbachicar — APPROVED** [review]
LGTM modulo one comment.

File: src/maliput_malidrive/test_utilities/assert_compare.h (hunk 1 above)
Comment (francocipollone) [reply src/maliput_malidrive/test_utilities/assert_compare.h:1]: test_utilities  (in malidrive) is not part of the public API

Comment (francocipollone) [reply src/maliput_malidrive/test_utilities/assert_compare.h]: Moved to test at d154041

**Pull request merged**